gapis/api/vulkan/extensions/khr_swapchain.api

```diff
@@ -266,7 +266,11 @@ cmd VkResult vkQueuePresentKHR(
   if (info.pWaitSemaphores != null) {
     wait_semaphores := info.pWaitSemaphores[0:info.waitSemaphoreCount]
     for i in (0 .. info.waitSemaphoreCount) {
-      Semaphores[wait_semaphores[i]].Signaled = false
+      if (Semaphores[wait_semaphores[i]].Signaled) {
+        Semaphores[wait_semaphores[i]].Signaled = false
+      } else {
+        // TODO: handle case when semaphore is not yet signaled
```

Comment:
This is going to be a weird, (and quite rare) case. In fact since no application I have ever seen uses events, I would posit that we will never actually hit this edge-case.

However we definitely should still handle it.
Would you mind adding a comment here describing the case in some detail (alternatively, opening a github issue, and linking to it).

The only "legal" time this can happen is if we signal a semaphore after a VkCmdWaitForEvents, and then submit this before the associated VkCmdSetEvents/VkSetEvent is hit. If you can see any other legal cases, please also include them (and let me know)

Reply:
Added bug (https://github.com/google/gapid/issues/1860) and link in the TODO comment.